### DISCUSSION TITLE
How to turn url back to the original url after submitting a form and regenerating content of a page with a successful message using Html::from?

### DISCUSSION THREAD

**Discussion by blueowlgreenfish** [tokio-rs/axum#1717] [Q&A] [answered]
I know that changing url is possible using `Redirect::to`, but I want to regenerate the same page with `Html::from` with a successful message with the original url name after user posts a form. `Redirect::to` can not be used to show the successful message after the form submission.
Right now, if a user posts a form, the page is regenerated with a successful message using `Html::from` and the url displays the name of the post action. I want the url to go back to the original url.
Could anyone tell me how to do this?

**Comment by davidpdrsn** [comment, marked as answer]
This is commonly called a "flash message". I have a crate for that here https://docs.rs/axum-flash

**Reply by blueowlgreenfish** [reply]
I didn't know about axum-flash. I was able to make it work using axum-flash and Redirect. Thank you!
